dataflow/pkg/scheduler/restoreobject/restoreobject_sched.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restoreobject
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/micro/go-log"
+	"github.com/micro/go-micro/client"
+	"github.com/opensds/multi-cloud/dataflow/pkg/db"
+	. "github.com/opensds/multi-cloud/dataflow/pkg/model"
+	osdss3 "github.com/opensds/multi-cloud/s3/proto"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+)
+
+var topicLifecycle = "restoreobject"
+var s3client = osdss3.NewS3Service("s3", client.DefaultClient)
+
+var mutext sync.Mutex
+
+// Get restore marker from each object from db, and schedule HEAD Object according to those values.
+func ScheduleRestoreObject() {
+	log.Log("[ScheduleRestoreObject] begin ...")
+
+	// Get bucket list.
+	listReq := s3.BaseRequest{Id: "test"}
+	listBucketRsp, err := s3client.ListBuckets(context.Background(), &listReq)
+	if err != nil {
+		log.Logf("[ScheduleRestoreObject]list buckets failed: %v.\n", err)
+		return
+	}
+
+	for _, v := range listBucketRsp.Buckets {
+		//For each bucket, list all the objects
+		getObjectInput := s3.ListObjectsRequest{Bucket: v.Name}
+		listObjectRsp, err := s3client.ListObjects(context.Background(), &getObjectInput)
+		if err != nil {
+			log.Logf("[ScheduleRestoreObject]list objects failed: %v.\n", err)
+			return
+		}
+		// For each object, get the restore marker rules, and schedule HEAD API
+		for _, object := range listObjectRsp.ListObjects {
+			restoreStatus := object.RestoreStatus
+			if restoreStatus == nil {
```

Comment:
Suggest to filter out objects which are not in the restoration process when in s3client.ListObjects, otherwise there will be too many objects.